src/commands/dev/lib/dev-builder.ts

```diff
@@ -373,10 +372,11 @@ export async function getBuildMatches(
     }
 
     // TODO: use the `files` map from DevServer instead of hitting the filesystem
```

Comment:
It would be nice to tackle this TODO instead of calling `getFiles()` again.

Reply:
Let's merge this PR for correctness and then do another PR where we measure the bottlenecks and fix performance issues.